src/append/rolling_file/policy/compound/mod.rs

```diff
@@ -155,3 +156,31 @@ impl Deserialize for CompoundPolicyDeserializer {
         Ok(Box::new(CompoundPolicy::new(trigger, roller)))
     }
 }
+
+#[cfg(not(test))]
+fn now_string(fmt: &str) -> String {
+    Local::now().format(fmt).to_string()
+}
+
+#[cfg(test)]
+use std::cell::RefCell;
+
+#[cfg(test)]
+thread_local! {
```

Comment:
Please add a comment on why this needs to be thread_local. A assume its some Send or Sync issue, why not use an Arc?

Reply:
When `set_mock_time` invoked,  RefCell can safely mutate on thread local variable.
We can also use `unsafe` to make `static mut`. Which solution is better?
```rust
#[cfg(test)]
static mut MOCK_TIME_STR: Option<String> = None;

#[cfg(test)]
fn now_string(_fmt: &str) -> String {
    unsafe {
        if let Some(time_str) = &MOCK_TIME_STR {
            time_str.to_string()
        } else {
            panic!("set mock now_string before use")
        }
    }
}

#[cfg(not(test))]
fn now_string(fmt: &str) -> String {
    Local::now().format(fmt).to_string()
}

#[cfg(test)]
fn set_mock_time(time: &str) {
    // MOCK_TIME_STR.with(|cell| *cell.borrow_mut() = Some(time.to_owned()));
    unsafe {
        MOCK_TIME_STR = Some(time.to_owned());
    }
}
```